src/pages/index.js

```diff
@@ -0,0 +1,393 @@
+const avatarForm = avatarModal.querySelector('.modal__form');
+const avatarSubmitBtn = avatarModal.querySelector('.modal__submit-btn');
+const avatarModalCloseBtn = avatarModal.querySelector('.modal__close-btn');
+const avatarInput = avatarModal.querySelector('#profile-avatar-input');
+
+// Delete Modal elements
+const deleteModal = document.querySelector('#delete-modal');
+const deleteForm = deleteModal.querySelector('.modal__form');
+const deleteModalCancelBtn = deleteModal.querySelector('#delete-cancel-btn');
+let selectedCard;
+let selectedCardId;
+
+function openModal(modal) {
+  modal.classList.add('modal_opened');
+  document.addEventListener('keydown', handleEscapeKey);
+}
+
+function closeModal(modal) {
+  modal.classList.remove('modal_opened');
+  document.removeEventListener('keydown', handleEscapeKey);
+}
+
+function handleEditFormSubmit(evt) {
+  evt.preventDefault();
+  // editModalSubmitBtn.textContent = 'Saving...';
+  // change text content to "Saving..."
+  const editModalSubmitBtn = evt.submitter;
+
+  setButtonText(editModalSubmitBtn, true, "Save", "Saving...");
+  // disable the button
+  api.editUserInfo({ name: editModalNameInput.value, about: editModalDescriptionInput.value })
+    .then((data) => {
+      console.log(data);
+      profileName.textContent = data.name;
+      profileDescription.textContent = data.about;
+      closeModal(editModal);
+    })
+    .catch(console.error)
+    .finally(() => {
+    setButtonText(editModalSubmitBtn, false);
+    
+    });
+}
+
+// todo - implement loading text for all other form submissions
+
+function handleAddCardFormSubmit(evt) {
+  evt.preventDefault();
+
+  const addCardModalSubmitBtn = evt.submitter;
+  setButtonText(addCardModalSubmitBtn, true, "Save", "Saving...");
+
+  const inputValues = { name: cardNameInput.value, link: cardLinkInput.value };
+  api
+    .addCard(inputValues)
+    .then((data) => {
+      console.log(data);
+
+      const cardElement = getCardElement(data);
+      cardsList.prepend(cardElement);
+      evt.target.reset();
+    }).catch(console.error)
+    .finally(() => {
+
+setButtonText(editModalSubmitBtn, false);
+    });
+  disableButton(cardSubmitBtn, settings);
+  closeModal(cardModal);
+
+}
+
+function handleAvatarSubmit(evt) {
+  evt.preventDefault();
+
+  const avatarSubmitBtn = evt.submitter;
+
+  setButtonText(avatarSubmitBtn, true, "Save", "Saving...");
+
+  console.log(avatarInput.value);
+  // const formData = new FormData(avatarForm);
+  // const avatarUrl = formData.get('profile-avatar-input');
+  api
+    .editAvatarInfo(avatarInput.value)
+    .then((data) => {
+      console.log(data);
+      // const avatarImageSrc = document.getElementById("profile-avatar");
+      avatarImageSrc.src = data.avatar;
+      closeModal(avatarModal);
+      evt.target.reset();
+  disableButton(avatarSubmitBtn, settings);
+    })
+    .catch(console.error)
+    .finally(() => {
+
+      setButtonText(editModalSubmitBtn,false);
```

Comment:
`editModalSubmitBtn`

* When I add a new card, there is an error https://snipboard.io/uCPXtg.jpg . And when I change the `avatar` as well https://snipboard.io/APngx6.jpg 

`editModalSubmitBtn` doesn't exist here. You should use `avatarSubmitBtn`


```js
    .finally(() => {
      setButtonText(avatarSubmitBtn, false);
    });
```

Reply:
My remark hasn't been corrected